### ISSUE TITLE
Commit cc9d27c50 broke the registration of certain ClassIds

### ISSUE THREAD

**Issue by orlandini** [labmec/neopz#33]
````Branch:```` RefactorGeom
````Commit:```` cc9d27c50f366590fedb89d17f6b6ae4770f01b3
````Parent commit:```` 9710e82e98adce05f6ab017537b53b6994d3a040
````Steps to reproduce:```` Run the target ExemploSimples

The changes introduced in commit cc9d27c50 are preventing the correct Read/Write operation that is supposed to happen in the ExemploSimples target. I have tracked the bug for a bit and I have noticed that it is somehow related to the ````TPZGeoElRefPattern<pzgeom::TPZGeoLine>```` class not being properly registered. 

Comparing the output of the program with the output obtained on the previous commit<sup>1</sup> with the compiler directive ````VerboseMode_Q```` set on ````TPZPersistenceManager.cpp```` has shown that several classes are not being registered. However, I was not able to identify precisely the reason for this behaviour.

<sup>1</sup> There is a simple initialisation that need to be corrected if one tries to run the target with the working directory in parent commit.

**Comment by orlandini** [comment]
I have written a [Minimal Non-Working Example](https://github.com/orlandini/StaticInitializationTest) in order to understand what is going on. I found out that with
```
add_library(libTest SHARED
        src/TExampleClass.cpp
        )
```

the output is
```
/pathToExecutable/StaticTest
TRestoreClass<TExampleClass<int> >::TRestoreClass() [T = TExampleClass<int>]

Process finished with exit code 0
```

However, if I make it a STATIC library, the output is:
```
/pathToExecutable/StaticTest

Process finished with exit code 0
```


@luizcmarques , would you have a hint on how to deal with this?

**Comment by luizcmarques** [comment]
At first glance, I have no clue as to why this is happening. Disregarding the output itself, are both targets StaticTest and libTest being generated? If so,  are their sizes reasonable if we consider type of linkage?  Maybe we could check them with command ldd (linux) or otool -L (macOS).

**Comment by philippedevloo** [comment]
I found a workaround. If the code in tpzgeoelrefpattern.cpp is not being used by the executable that links with the library, the explicit instantiations are also not be generated. I solved the problem by putting the instantiations in the tpzgeoelrefpattern.h.h file whose code is always being called.

I put some logging statements in the code to track this kind of behavior in the future. The write code will now break if it is 
writing a class object with ClassID that hasn't been registered (in debug mode)
